subsys: spm: Add PWM0 and PWM1

jtguggedal opened the pull request:
This patch adds PWM0 and PWM1 to SPM so that they can be set
as non-secure using Kconfig.

Signed-off-by: Jan Tore Guggedal <jantore.guggedal@nordicsemi.no>

hakonfam commented on subsys/spm/Kconfig:
```diff
@@ -131,6 +131,13 @@ config SPM_NRF_SAADC_NS
+
+config SPM_NRF_PWM1_NS
+	bool "PWM1 is Non-Secure"
+
```
All peripherals which can be configured as non-secure, should have non-secure as default.
Unless there is a valid reason for not doing so (e.g. UARTE2 reserved for mcuboot usage).

hakonfam commented on subsys/spm/spm.c:
```diff
@@ -334,6 +334,8 @@ static void spm_config_peripherals(void)
 		PERIPH("NRF_TIMER2", NRF_TIMER2_S, CONFIG_SPM_NRF_TIMER2_NS),
 		PERIPH("NRF_SAADC", NRF_SAADC_S, CONFIG_SPM_NRF_SAADC_NS),
+		PERIPH("NRF_PWM0", NRF_PWM0_S, CONFIG_SPM_NRF_PWM0_NS),
+		PERIPH("NRF_PWM1", NRF_PWM1_S, CONFIG_SPM_NRF_PWM1_NS),
```
Maybe we should add the rest of the PWM's as well? While we are at it.

jtguggedal replied:
Done

ioannisg commented on subsys/spm/spm.c:
```diff
@@ -334,6 +334,8 @@ static void spm_config_peripherals(void)
 		PERIPH("NRF_TIMER1", NRF_TIMER1_S, CONFIG_SPM_NRF_TIMER1_NS),
 		PERIPH("NRF_TIMER2", NRF_TIMER2_S, CONFIG_SPM_NRF_TIMER2_NS),
 		PERIPH("NRF_SAADC", NRF_SAADC_S, CONFIG_SPM_NRF_SAADC_NS),
+		PERIPH("NRF_PWM0", NRF_PWM0_S, CONFIG_SPM_NRF_PWM0_NS),
```
NRF_PWMx should be defined, AFAICS.

jtguggedal replied:
Looks like I need the `_S` for peripherals that are not set to `status = "ok"` in DT. Only `PWM0` is enabled for the board, so I'll have to keep `_S` for the remaining instances. I'll remove from the timers, TWI and SPI.

UARTE2 is disabled because it's reserved for MCU boot, according to @hakonfam .

hakonfam commented:
> * I am also wondering @hakonfam @jtguggedal why UARTE2 is the only one with default n?
> 

I guess this is because of some MCUboot serial recovery functionality?

jtguggedal commented:
@ioannisg @hakonfam Updated according to your comments: Added all PWM instances and set them to `default y`, and removed the `_S` suffix where possible.

ioannisg commented on subsys/spm/spm.c:
```diff
@@ -328,12 +328,16 @@ static void spm_config_peripherals(void)
+		PERIPH("NRF_SAADC", NRF_SAADC, CONFIG_SPM_NRF_SAADC_NS),
+		PERIPH("NRF_PWM0", NRF_PWM0, CONFIG_SPM_NRF_PWM0_NS),
+		PERIPH("NRF_PWM1", NRF_PWM1_S, CONFIG_SPM_NRF_PWM1_NS),
+		PERIPH("NRF_PWM2", NRF_PWM2_S, CONFIG_SPM_NRF_PWM2_NS),
```
Like this it looks really messy; PWM0 exists but PWM1,2,3 require the _S.
I'd like a clean solution, here, i.e. define the PWMs as "ok" and use the standard macros

jtguggedal replied:
I agree it doesn't look good, but defining the instances "ok" is an upstream fix as far as I can see.

ioannisg replied:
@hakonfam  @jtguggedal  Is there any dts overlay for nrf9160_pca10090 when we build the SPM? If so, these could be defined there as "ok". The justification would be that "the SPM needs them to set them to non-secure" :)

jtguggedal replied:
If it's possible to pass an overlay to the SPM build, I agree that's a better solution. Do you know if that already happens or is possible, @hakonfam ?

jtguggedal replied:
Looks like it's the `SPM` sample that's built, so I'll add the peripherals in the already existing overlay there.

hakonfam commented:
> @ioannisg @hakonfam Updated according to your comments: Added all PWM instances and set them to `default y`, and removed the `_S` suffix where possible.

Please update the spm .rst to list the added peripherals.

jtguggedal commented:
Added PWM1-3 to overlay and added the peripherals to SPM docs.

hakonfam commented:
I just remembered that this should go in the changelog as well. Please put a sentence there about this change (as it is breaking API - PWM used to be secure, is now non-secure).

ioannisg commented:
> 
> 
> I just remembered that this should go in the changelog as well. Please put a sentence there about this change (as it is breaking API - PWM used to be secure, is now non-secure).

@jtguggedal FYI

jtguggedal commented:
> I just remembered that this should go in the changelog as well. Please put a sentence there about this change (as it is breaking API - PWM used to be secure, is now non-secure).

@hakonfam Would that be the changelog on the GitHub Wiki (https://github.com/NordicPlayground/fw-nrfconnect-nrf/wiki/Changelog) ?
If so, I think I need some assistance from @ru-fu on how to make PRs to the wiki/changelog.

hakonfam commented:
> @hakonfam Would that be the changelog on the GitHub Wiki (https://github.com/NordicPlayground/fw-nrfconnect-nrf/wiki/Changelog) ?
> If so, I think I need some assistance from @ru-fu on how to make PRs to the wiki/changelog.

Yes. AFAIK the author of the change will provide the update to that document. My assumption is that this document will be reviewed before release. Hence you should provide the "initial" version. There is no PR, my understanding is that you simply go in and edit the document. If this is not the correct procedure you can blame it on me :)

jtguggedal commented:
@hakonfam Thanks, I added the following: `PWM0-3 added as non-secure by default`.
@ru-fu Let me know if I should change it.
